### PR TITLE
update cypress and unit test documentation

### PR DESCRIPTION
**Pull request by christianvogt**
Closes:
https://issues.redhat.com/browse/RHOAIENG-841 
https://issues.redhat.com/browse/RHOAIENG-840

## Description

Added new `testing.md` markdown for all testing related documentation.
Updated various files that contained references to testing.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+```
+
+### Page Objects
+
+Page Objects are TypeScript objects / classes used to encapsulate all the selectors and functions related to a particular page or UI element. The functions exposed by a page object use the following prefixes:
+
+- `visit`: Navigate directly to the URL associated with the UI element.
+- `navigate`: Navigate to a page as defined by a link in the left navigation.
+- `find`:
+  - Without prefix (`find()`): Eestablish a scope for command chaining.
```
Comment (manaswinidas) [inline docs/testing.md:194]: Typo: Eestablish -> Establish

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+Jest is used as the unit test runner. Unit tests should be written for all utility and hook functions; React Components do not require unit tests.
+
+Running unit tests:
+```bash
+npm run test:unit
+```
+
+### Structure
+
+Test files must be consistently named based on the name of the source code file being tested. Test files must use the `.spec.ts` file extension and be located in the `__test__` directory adjacent the source code file.
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply docs/testing.md]: Oh that's a bad typo lol

**Comment by christianvogt** [comment]
Addressed comments. Ready for review again.

**Review by manaswinidas — APPROVED** [review]
/lgtm

**Review by pnaik1 — CHANGES_REQUESTED** [review]

File: docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+- `get`: Returns another page object that scopes selectors within the context of this object.
+  - Eg. `getRow(...)` would return an object for working with a specific row of a table.
+
+_All selectors should reside in page objects and not individual tests._
+
+### Selectors and Test IDs
+
+The primary method for selecting elements on a page is through the use of test IDs using the `data-testid` attribute. Update the frontend code as needed when a new part of the UI needs to be selected by a test. Where test IDs cannot be applied, role based selectors may be used. For example when attempting to select a link is derived from data.
+
+- `cy.findByTestId('<test-id')`
```
Comment (ppadti) [inline docs/testing.md:210]: Nit:
```suggestion
- `cy.findByTestId('<test-id>')`
```

File: docs/testing.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
+# Testing
+
+Running all tests:
+```bash
+npm run test
+```
+
+## Unit Tests
+
+Jest is used as the unit test runner. Unit tests should be written for all utility and hook functions; React Components do not require unit tests.
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply docs/testing.md:10]: I think the semi colon is appropriate here because the two clauses, while independent, are part of the same idea.

**Pull request merged**